test/jdk/javax/net/ssl/DTLS/DTLSSignatureSchemes.java

```
@@ -0,0 +1,134 @@
+        Security.setProperty("jdk.tls.disabledAlgorithms", "");
+
+        runTest(new String[] {
+                        "ecdsa_secp256r1_sha256",
+                        "ed25519"
+                },
+                new String[] {
+                        "ecdsa_secp256r1_sha256",
+                        "ed25519"
+                },
+                false);
+        runTest(new String[] {
+                        "ecdsa_secp256r1_sha256"
+                },
+                new String[] {
+                        "ecdsa_secp256r1_sha256"
+                },
+                false);
+        runTest(null,
+                new String[] {
+                        "ecdsa_secp256r1_sha256"
+                },
+                false);
+        runTest(new String[] {
+                        "ecdsa_secp256r1_sha256"
+                },
+                null,
+                false);
+        runTest(new String[0],
+                new String[] {
+                        "ecdsa_secp256r1_sha256"
+                },
+                true);
+        runTest(new String[] {
+                        "ecdsa_secp256r1_sha256"
+                },
+                new String[0],
+                true);
+        runTest(new String[] {
+                        "ecdsa_secp256r1_shaNA"
+                },
+                new String[] {
+                        "ecdsa_secp256r1_sha256"
+                },
+                true);
+    }
+
+    private static void runTest(String[] serverSignatureSchemes,
+                                String[] clientSignatureSchemes,
+                                boolean exceptionExpected) throws Exception {
+        DTLSSignatureSchemes testCase = new DTLSSignatureSchemes(
+                serverSignatureSchemes, clientSignatureSchemes);
+        try {
+            testCase.runTest(testCase);
+            if (exceptionExpected) {
+                throw new RuntimeException("Unexpected success!");
```

The catch block on line 127 will end up catching this exception and swallowing it, and the test will incorrectly pass.

---

Good catch!